pulp_rpm/app/models/repository.py

```diff
@@ -232,42 +227,6 @@ class RpmRepository(Repository):
     repo_gpgcheck = models.IntegerField(default=0, choices=GPGCHECK_CHOICES)
     sqlite_metadata = models.BooleanField(default=False)
 
-    def new_version(self, base_version=None):
-        """
-        Create a new RepositoryVersion for this Repository.
-
-        Creation of a RepositoryVersion should be done in a RQ Job.
-
-        Args:
-            repository (pulpcore.app.models.Repository): to create a new version of
-            base_version (pulpcore.app.models.RepositoryVersion): an optional repository version
-                whose content will be used as the set of content for the new version
-
-        Returns:
-            pulpcore.app.models.RepositoryVersion: The Created RepositoryVersion
-
-        """
-        with transaction.atomic():
-            latest_version = self.versions.latest()
-            if not latest_version.complete:
-                latest_version.delete()
-
-            version = RepositoryVersion(
-                repository=self, number=int(self.next_version), base_version=base_version
-            )
-            version.save()
-
-            if base_version:
-                # first remove the content that isn't in the base version
-                version.remove_content(version.content.exclude(pk__in=base_version.content))
-                # now add any content that's in the base_version but not in version
-                version.add_content(base_version.content.exclude(pk__in=version.content))
-
-            if Task.current() and not self.sub_repo:
-                resource = CreatedResource(content_object=version)
-                resource.save()
```

Comment:
I checked the pulpcore version to make sure we were now safe to delete this, and it turned out that we are, but the RPM version here was missing a cache invalidation event.